File: benchmark/ruby/prime_count.rb
start = Time.now.to_f * 1000
count = 0
(2..100000).each do |n|
    is_p = true
    if n < 2 then is_p = false
    elsif n == 2 then is_p = true
    elsif n % 2 == 0 then is_p = false
    else
        i = 3
        while i * i <= n
            if n % i == 0
                is_p = false
                break
            end
            i += 2
        end
    end
    count += 1 if is_p
end
finish = Time.now.to_f * 1000
puts "Primes: #{count}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/nodejs/prime_count.js
const start = Date.now();
let count = 0;
for (let n = 2; n <= 100000; n++) {
    let is_p = true;
    if (n < 2) is_p = false;
    else if (n === 2) is_p = true;
    else if (n % 2 === 0) is_p = false;
    else {
        for (let i = 3; i * i <= n; i += 2) {
            if (n % i === 0) {
                is_p = false;
                break;
            }
        }
    }
    if (is_p) count++;
}
const end = Date.now();
console.log(`Primes: ${count}`);
console.log(`Time: ${end - start}ms`);

